txclib/project.py

```diff
@@ -370,9 +368,15 @@ def _get_pseudo_file(self, slang, resource, file_filter):
         pseudo_file = file_filter.replace('<lang>', '%s_pseudo' % slang)
         return native_path(pseudo_file)
 
+    def _slug_with_branch(self, resource_slug, branch):
+        """ Return the resource slug prefixed by the branch name.
+        """
+        return '{branch}--{resource}'.format(branch=branch,
+                                             resource=resource_slug)
+
     def pull(self, languages=[], resources=[], overwrite=True, fetchall=False,
              fetchsource=False, force=False, skip=False, minimum_perc=0,
```

Comment:
Suggestion for a refactoring:
In order to decrease the number of arguments passed in the pull/push methods, we could introduce a config object with related args like fetchall, fetchsource, force, skip, mode, etc.
(1) [Long Parameter list code smell](https://refactoring.guru/smells/long-parameter-list)
(2) [Introduce parameter object technique](https://refactoring.guru/introduce-parameter-object)